pymc3/backends/base.py

```diff
@@ -552,17 +554,27 @@ def merge_traces(mtraces):
 
     Raises
     ------
-    A ValueError is raised if any traces have overlapping chain numbers.
+    A ValueError is raised if any traces have overlapping chain numbers,
+    or if chains are of different lengths.
 
     Returns
     -------
     A MultiTrace instance with merged chains
     """
+    if len(mtraces) == 0:
+        raise ValueError("Cannot merge an empty set of traces.")
     base_mtrace = mtraces[0]
+    chain_len = len(base_mtrace)
+    # check base trace
+    if any((len(st) != chain_len
```

Comment:
missing `:`?

Reply:
No, that's just an oddity that comes because the iteration expression inside the `any` is so long.  The colon is on line 570.

Reply:
I'm going to flout pylint's "line too long" and put these two lines back together for readability.